src/api/Nhs.Appointments.Api/Json/JsonToObjectSchemaValidation.cs

```diff
@@ -81,22 +81,28 @@ private static List<ErrorMessageResponseItem> CheckObject(JsonElement json, Type
             if (objType == typeof(Object)) 
                 return errors;
 
+            var jsonProperties = objType.GetProperties().Select(p => (p, p.GetCustomAttribute<JsonPropertyAttribute>()));
+            var requiredProperties = jsonProperties.Where(jp => jp.Item2?.Required == Required.Always).Select(p => p.Item1.Name).ToList();
+
             while (objectEnumerator.MoveNext())
             {
                 var jsonProperty = objectEnumerator.Current;
-                var objProperty = GetProperty(jsonProperty.Name, objType);
+                var objProperty = GetProperty(jsonProperty.Name, objType);                
```

Comment:
Not a comment on your PR really, but this whitespace addition just reminds me that we still haven't set up auto formatting rules for the backend like we have for the frontend. I've also been noticing it with line length recently when I kept expecting longer lines to wrap
I'll mention it in refinement